lib/MPIPreferences/src/MPIPreferences.jl

```diff
@@ -137,6 +147,11 @@ function use_system_binary(;
     if isnothing(abi)
         abi = identify_abi(libmpi)
     end
+    if isnothing(abi_file)
+        abi_file = ""
+    else
+        abi_file = abspath(abi_file)
+    end
```

Comment:
i'd leave it as `nothing` if it is not set.

maybe a slightly longer name (e.g. `abi_source_file`?)

also, you need to store it via `set_preferences!`

Reply:
> i'd leave it as `nothing` if it is not set.

That won't work, since `Nothing` is not a valid TOML type:
```julia
julia> TOML.print(Dict{String,Any}("abi_file" => nothing))
ERROR: type `Nothing` is not a valid TOML type, pass a conversion function to `TOML.print
```
This is also the reason why used the empty string as a sentinel. I could also use `false` if you'd prefer that.

> maybe a slightly longer name (e.g. `abi_source_file`?)

Makes sense, I will do that.

> also, you need to store it via `set_preferences!`

Thanks, I seem to have forgotten to commit that part 🙈

Reply:
> maybe a slightly longer name (e.g. `abi_source_file`?)

Done in 162653a.

Reply:
> That won't work, since `Nothing` is not a valid TOML type:

Preferences.jl supports it, and it is what is returned when value is not set:
```
julia> isnothing(load_preference(MPIPreferences, "abi_file"))
true
```